Compass/ParseServer.swift

```diff
@@ -82,23 +149,122 @@ class PServer {
                     print("object doesnt exist")
                     return
                 }
-
-            newUserData["Location"] = PFGeoPoint(location: uData.location)
+            newUserData["Location"] = PFGeoPoint(location: location)
+            newUserData["UserName"] = PFUser.current()?.username
             newUserData.saveInBackground(block: { (sucess, error) in
                 guard error == nil else {
                     print(error ?? "Failed to update data")
                     return
                 }
-
             })
         })
     }
+    
+    func fetchUserList(completition: @escaping ([String]?, Error?) -> Void) {
+        let query = PFUser.query()
+        query?.findObjectsInBackground(block: { (objects, error) in
+            var result = [String]()
+            guard error == nil else {
+                completition(nil, error)
+                return
+            }
+            if let users = objects {
+                for object in users {
+                    if let user = object as? PFUser {
+                        if user.username != PFUser.current()?.username {
+                            result.append(user.username!)
+                        }
+                    }
+                }
+                completition(result, nil)
```

Comment:
Similar to above, the completion won't be called if `if let users = objects {` fails.